content/en/docs/tasks/administer-cluster/cpu-management-policies.md

```diff
@@ -303,6 +304,21 @@ policy option is not compatible with `TopologyManager` `single-numa-node`
 policy and does not apply to hardware where the number of sockets is greater
 than number of NUMA nodes.
 
+
+If the `distribute-cpus-across-cores` policy option is specified, the static policy
+will attempt to allocate virtual cores (hardware threads) across different physical cores.
+By default, the `CPUManager` tends to pack cpus onto as few physical cores as possible,
+which can lead to contention among cpus on the same physical core and result
+in performance bottlenecks. By enabling the `distribute-cpus-across-cores` policy,
+the static policy ensures that cpus are distributed across as many physical cores
+as possible, reducing the contention on the same physical core and thereby
+improving overall performance. However, it is important to note that this strategy
+might be less effective when the system is heavily loaded. Under such conditions,
+the benefit of reducing contention diminishes. Conversely, default behavior
+can help in reducing inter-core communication overhead, potentially providing
+better performance under high load conditions.
```

Comment:
While the virtual cores are backed by more physical cores, the application can potentially get performance boost thanks to more L2, L3 caches to use.
However, these physical cores may come from different sockets.
The assumption of better performance is only true when the following factors are met:

- there are heavy inter-thread communication in the application;
- the load is not very high, or else you may want to use more cores rather than fewer,
  which is the newly added policy.

Am I understanding this correctly?

Reply:
Exactly. Our benchmarks show that this approach works well in low-load scenarios. We employ this strategy for serverless workloads, where it's uncommon for all applications to be busy simultaneously. This policy performs effectively in such environments.

We've also tested the potential impact of cross-socket issues and found no significant difference for our application. However, other applications may experience varying results. In the alpha version, we did not explicitly consider whether CPUs were across sockets or not. In future versions, we plan to improve this aspect and clarify the behavior.